VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,215 @@
+                new StubAuthorData(null, null, creationDate), now,
+                answerPrefix + " " + relativeTime);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, null), now,
+                answerPrefix + " " + outputAuthor);
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, null), now,
+                answerPrefix + " " + outputAuthorLbl);
+
+        // Empty Case
+        assertSetAuthorAttributionText(textView,
+                DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, null), now,
+                null);
+    }
+
+    private void assertSetAuthorAttributionText(TextView textView,
+                                                DiscussionTextUtils.AuthorAttributionLabel type,
+                                                IAuthorData input,
+                                                long now, String expectedOutput) {
+        final Runnable listener = Mockito.mock(Runnable.class);
+        DiscussionTextUtils.setAuthorAttributionText(textView, type, input, now, listener);
+        if (expectedOutput == null) {
+            assertTrue(textView.getVisibility() == View.GONE);
+        } else {
+            String output = textView.getText().toString();
+            assertEquals(expectedOutput, output);
+            if (!input.isAuthorAnonymous()) {
+                // Test whether author span is clickable or not
+                int start = output.indexOf(input.getAuthor());
+                int end = start + input.getAuthor().length();
+                SpannableString text = (SpannableString) textView.getText();
+                StyleSpan[] styleSpans = text.getSpans(start, end, StyleSpan.class);
+                ForegroundColorSpan[] colorSpans = text.getSpans(start, end, ForegroundColorSpan.class);
+                if (config.isUserProfilesEnabled()) {
+                    // Check if the author text is Bold
+                    assertEquals(1, styleSpans.length);
+                    assertEquals(start, text.getSpanStart(styleSpans[0]));
+                    assertEquals(end, text.getSpanEnd(styleSpans[0]));
+                    assertEquals(Typeface.BOLD, styleSpans[0].getStyle());
+
+                    // Check if the correct foreground color is set
+                    assertEquals(1, colorSpans.length);
+                    assertEquals(start, text.getSpanStart(colorSpans[0]));
+                    assertEquals(end, text.getSpanEnd(colorSpans[0]));
+                    assertEquals(context.getResources().getColor(R.color.edx_brand_primary_base),
+                            colorSpans[0].getForegroundColor());
+
+                    // Check if the whole text view is clickable
+                    textView.callOnClick();
```

Comment:
Since these methods were introduced in API 15, you should add a `@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1)` annotation to the method definition to get rid of the warning.